### ISSUE TITLE
Discussion: How to design network calls with the same lifecycle as of controller?

### ISSUE THREAD

**Issue by bangarharshit** [bluelinelabs/Conductor#391]
I want to design a network call with same the lifecycle as of controller (start with push and ends when the controller is popped).

The examples in [RxLifecycle 2](https://github.com/bluelinelabs/Conductor/blob/develop/demo/src/main/java/com/bluelinelabs/conductor/demo/controllers/RxLifecycle2Controller.java) module suggests using the same in `constructor` and dispose it in `onDestroy`.

The example leads to [`this`](https://www.ibm.com/developerworks/library/j-jtp0618/index.html) leak another thread which is a bug. Even for the same thread, it is an anti-pattern. In `Activity` and `Fragment` we have onCreate for the same.

Is there any recommended pattern for the same?

**Comment by bangarharshit** [comment]
It is in part duplicate of - https://github.com/bluelinelabs/Conductor/issues/34. The other part `leaking this` remains an issue. Is there a reason why you are doing it (maybe I am missing something)? 

Also, do you think push/pull solves the issues with onCreate as mentioned in the above issue?.

View attach/detach can be very different from the controller lifecycle themselves (in Rx multicasting can help you bridge the two).

**Comment by SamYStudiO** [comment]
+1 for onCreate
Another use case:
Actually we need to create viewmodel from `onContextAvailable` but if we want for example send `savedInstanceState` bundle from `onRestoreInstanceState` we cannot do it since `onRestoreInstanceState` is called before `onContextAvailable` and our viewmodel in not yet created.
By the way i can't figure out why  #34 has been closed?

**Comment by hijamoya** [comment]
Now I need to do things(like injection) twice one in `onRestoreInstanceState` and one in `onContextAvailable ` and use a flag to keep that code block run only once.

We need a` onCreate()` which called before `onRestoreInstanceState`
